pkg/openmetadata-connector-core/connector_helper.go

```diff
@@ -52,94 +52,140 @@ func tagColumn(ctx context.Context, c *client.APIClient, columns []client.Column
 	return columns
 }
 
-func (s *OpenMetadataAPIService) prepareOpenMetadataForFybrik() bool {
+// transform variable `tags` into an array.
+// traverse `tags` and create OM tags in the `categoryName` category
+func (s *OpenMetadataAPIService) addTags(ctx context.Context, c *client.APIClient,
+	categoryName string, tags interface{}) {
+	tagsArr, ok := utils.InterfaceToArray(s.logger, tags)
+	if !ok {
+		s.logger.Warn().Msg("Malformed tag information")
+		return
+	}
+	for i := range tagsArr {
+		func() {
+			tagMap, ok := tagsArr[i].(map[interface{}]interface{})
+			if ok {
+				name, ok1 := tagMap[Name]
+				description, ok2 := tagMap[Description]
+				if ok1 && ok2 {
+					_, r, err := c.TagsApi.CreatePrimaryTag(ctx, categoryName).
+						CreateTag(*client.NewCreateTag(description.(string), name.(string))).Execute()
+					if err != nil {
+						s.logger.Trace().Msg("Failed to create Tag. Maybe it already exists.")
+					} else {
+						r.Body.Close()
+						return
+					}
+				} else {
+					s.logger.Warn().Msg(fmt.Sprintf("malformed tag information. cannot cast %T to map[interface{}]interface{}", tagsArr[i]))
+				}
+			}
+		}()
+	}
+}
+
+func (s *OpenMetadataAPIService) PrepareOpenMetadataForFybrik() bool { //nolint
```

Comment:
Theoretically OMD can be initialized from a command line and after that method can be called again. Or the same catalog can be shared between several Fybriks (OMD connectors), or after restart of an OMD connector, so we have to be ready to run the method several times. 

Will be worth to add a test to run the method several times.

Reply:
I suggest that we postpone until we have end-to-end tests (rather than unit tests based on mock servers)